vumi_http_proxy/test/test_http_proxy.py

```diff
@@ -0,0 +1,96 @@
+from twisted.internet import reactor
+from twisted.internet.defer import inlineCallbacks, Deferred, returnValue
+from twisted.internet.endpoints import serverFromString, clientFromString
+from twisted.trial import unittest
+from twisted.test import proto_helpers
+from twisted.web.client import ProxyAgent, readBody
+
+from vumi_http_proxy.http_proxy import ProxyFactory
+
+from .helpers import HttpTestServer, DEFAULT_TIMEOUT
+
+timeout = DEFAULT_TIMEOUT
+
+
+class ProxyToLocalServerTestCase(unittest.TestCase):
```

Comment:
I prefer using names like `TestProxyToLocalServer` for test classes and using `SomethingTestCase` for a base class that other test classes inherit. For example, `VumiTestCase` is a subclass of `TestCase` that adds various vumi-specific functionality and then `TestThisVumiThing` would subclass `VumiTestCase` and contain test methods.

This is mostly just a naming convention, but there are test runners that use class and function name prefixes to collect tests.